app/assets/javascripts/add_date_field.js.coffee

```diff
@@ -0,0 +1,32 @@
+jQuery ->
+  $('#add_new_date_type').on("click", (event) ->
+    event.preventDefault()
+    
+    #Variable assignment and necessary string manipulation
+    notice = '<div class="alert alert-info alert-dismissible" role="alert"><button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button> That date field already exists </div>'
+    type = $('#new_date_type').val()
+    type_string = type_manipulation(type)
+    date_created = $(".form-group.generic_file_date_created")
+    html = html_manipulation(date_created.clone(), type, type_string)
+
+    html.find("ul.listing li:not(:last-child)").remove()
```

Comment:
I have reason to believe this is specific enough to only target date field stuff. This picks out the listing from the html we parsed out, which is a copy of the 'generic_file_date_created' section of the form, of which there is only one.